lib/url.js

```diff
@@ -693,23 +700,23 @@ Url.prototype.format = function format() {
   // unless they had them to begin with.
   if (this.slashes || slashedProtocol.has(protocol)) {
     if (this.slashes || host) {
-      if (pathname && pathname.charCodeAt(0) !== CHAR_FORWARD_SLASH)
+      if (pathname && pathname[0] !== '/')
         pathname = '/' + pathname;
       host = '//' + host;
     } else if (protocol.length >= 4 &&
-               protocol.charCodeAt(0) === 102/* f */ &&
-               protocol.charCodeAt(1) === 105/* i */ &&
-               protocol.charCodeAt(2) === 108/* l */ &&
-               protocol.charCodeAt(3) === 101/* e */) {
+               protocol[0] === 'f' &&
+               protocol[1] === 'i' &&
+               protocol[2] === 'l' &&
+               protocol[3] === 'e') {
```

Comment:
no because isFileProtocol is also accepting ':' at the end of the string but here according to the comment it is about adding ':' at the end of the protocol if it is missing.

Reply:
i guess you could use it after the length check

Reply:
for performance reasons, i think this is better